### PR TITLE
feat(website): provide instruction how to report errors

### LINKED ISSUE #201: Add tracable error info on 500 error page
In #196 we added a generic error page. Add more information to that page, so we can trace the error. A possible solution could be

> I'm thinking whether we should add some more details to this page. If users are supposed to create an issue, it's valuable for us to know what actually went wrong.

> https://docs.astro.build/en/basics/astro-pages/#error
> Maybe we can extract something from the error prop?

> An alternative would be logging the error prop and showing the user some tracing info that we can use to retrieve the error from logs. That could be a request id or an "error id" (a UUID that we generate here and log it with the error). Anyway, we need

> -   to tell users: "copy paste this to the issue so that we can reproduce the issue"
> -  a way to retrieve the error information using this

> It's probably worth it's own issue (along with proper logging in general).

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #201

### Summary

If something unexpected happens,
- show an error id to the user
- tell the user to report that id
- log the error with that id so that we can find it in the logs

A follow-up task is providing more useful error messages to users: https://github.com/GenSpectrum/dashboards/issues/256

### Screenshot

When an error occurs in server-side code:
grafik

When an error occurs in client-side code:
grafik

grafik

### CODE REVIEW

File: website/src/components/ErrorReportInstruction.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                </div>
+            </dialog>
+        </>
+    );
+}
+
+export function ErrorReportInstruction({ errorId, currentUrl }: { errorId: string; currentUrl: string }) {
+    return (
+        <>
+            <h3 className='mb-4 text-lg font-bold'>Want to report an error?</h3>
```
Comment (JonasKellerer) [inline website/src/components/ErrorReportInstruction.tsx:37]:
```suggestion
            <h3 className='mb-4 text-lg font-bold'>Help us fixing the issue</h3>
```

File: website/src/components/ErrorReportInstruction.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            <p>
+                Please create an issue on{' '}
+                <a className='text-primary underline' href='https://github.com/GenSpectrum/dashboards'>
+                    GitHub
+                </a>
+                . Describe what you did to trigger the error and include the following information so that we can track
+                down the issue:
+                <ul className='ml-6 mt-2 list-disc'>
```
Comment (JonasKellerer) [inline website/src/components/ErrorReportInstruction.tsx:45]: Can we add a "Copy" button, which copies the relevant info to the clipboard?

File: website/src/components/ErrorReportInstruction.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            modalRef.current.showModal();
+        }
+    };
+
+    return (
+        <>
+            <p>
+                The problem persists?{' '}
+                <button className='link' onClick={openModal}>
+                    Learn how to report an issue.
```
Comment (JonasKellerer) [inline website/src/components/ErrorReportInstruction.tsx:17]:
```suggestion
                    Help us fixing the issue.
```

File: website/src/components/subscriptions/create/SubscriptionsCreate.tsx [hunk 4]
```diff
@@ -48,12 +50,20 @@ export function SubscriptionsCreateInner({
...
-            toast.error('Failed to create subscription', {
-                position: 'bottom-left',
-                autoClose: false,
+            const errorId = uuidv4();
+            logger.error(`Failed to create a new subscription: ${getErrorLogMessage(error)}`, {
+                errorId,
             });
+            toast.error(
+                <>
+                    <p className='mb-2'>Failed to create subscription. Please try again later.</p>
```
Comment (JonasKellerer) [inline website/src/components/subscriptions/create/SubscriptionsCreate.tsx:59]:
```suggestion
                    <p className='mb-2'>We could not create your subscription. Please try again later.</p>
```

File: website/src/components/subscriptions/overview/SubscriptionEntry.tsx [hunk 5]
```diff
@@ -107,12 +109,20 @@ function MoreDropdown({
...
-            toast.error(`Failed to delete subscription "${subscription.name}"`, {
-                position: 'bottom-left',
-                autoClose: false,
+            const errorId = uuidv4();
+            logger.error(`Failed to delete subscription with id '${subscription.id}': ${getErrorLogMessage(error)}`, {
+                errorId,
             });
+            toast.error(
+                <>
+                    <p className='mb-2'>Failed to delete subscription "{subscription.name}". Please try again later.</p>
```
Comment (JonasKellerer) [inline website/src/components/subscriptions/overview/SubscriptionEntry.tsx:118]:
```suggestion
                    <p className='mb-2'>We could not delete your subscription "{subscription.name}". Please try again later.</p>
```

File: website/src/pages/500.astro [hunk 6]
```diff
@@ -11,18 +15,14 @@ interface Props {
...
 const { error } = Astro.props;
 
-getInstanceLogger('500ErrorPage').error(`An error occurred: ${getErrorLogMessage(error)}`);
+const errorId = uuidv4();
+getInstanceLogger('500ErrorPage').error(`An error occurred: ${getErrorLogMessage(error)}`, { errorId });
 ---
 
 <BaseLayout title='GenSpectrum Dashboards'>
     <PageContainer>
         <PageHeadline>Oops! An error occurred...</PageHeadline>
```
Comment (JonasKellerer) [inline website/src/pages/500.astro:24]:
```suggestion
        <PageHeadline>Oops! Something went wrong...</PageHeadline>
```

File: website/src/pages/500.astro [hunk 7]
```diff
@@ -11,18 +15,14 @@ interface Props {
...
-        <div class='mb-2'>We are sorry for the inconvenience. Please try again.</div>
-        <div>
-            If the problem persists, please create an issue on <a
-                class='text-primary underline'
-                href='https://github.com/GenSpectrum/dashboards'>GitHub</a
-            >, describing the error.
-        </div>
+        <div class='mb-2'>We are sorry for the inconvenience. Please try again later.</div>
```
Comment (JonasKellerer) [inline website/src/pages/500.astro:25]:
```suggestion
        <div class='mb-2'>We are sorry for the inconvenience. Please try again later.</div>
        <div>You have seen this message multiple times?</div>
```

Comment (JonasKellerer) [reply website/src/pages/500.astro:25]: I would put the `ErrorReportInstruction inside a collapse and make the "You have seen this message multiple times?" the handle.
This way the user is not exposed to too much information.

**Review by JonasKellerer — CHANGES_REQUESTED** [review]
Nice. Good to have. :)
I would adjust the error messages slightly, especially when we don't do the user specific error messages before the public launch.
And I would put the "Report error" inside a collapse on server side, so the user is not exposed to too much information.

File: website/src/components/ErrorReportInstruction.tsx (hunk 2 above)
Comment (fengelniederhammer) [reply website/src/components/ErrorReportInstruction.tsx:45]: I created #264

**Review by JonasKellerer — APPROVED** [review]
LGTM

**Pull request merged**